server/proxy/session.go

```diff
@@ -451,14 +459,33 @@ func (this *Session) handlePacket(packet packet.Packet) (err error) {
 			if err != nil {
 				return
 			}
-			var verifyToken []byte
-			verifyToken, err = rsa.DecryptPKCS1v15(cryptoRand.Reader, this.server.privateKey, loginEncryptResponse.VerifyToken)
-			if err != nil {
-				return
-			}
-			if bytes.Compare(this.verifyToken, verifyToken) != 0 {
-				err = errors.New("Verify token does not match")
-				return
+			if loginEncryptResponse.VerifyToken != nil {
+				var verifyToken []byte
+				verifyToken, err = rsa.DecryptPKCS1v15(cryptoRand.Reader, this.server.privateKey, loginEncryptResponse.VerifyToken)
+				if err != nil {
+					return
+				}
+				if bytes.Compare(this.verifyToken, verifyToken) != 0 {
+					err = errors.New("Verify token does not match")
+					return
+				}
+			} else {
+				var parsedKey interface{}
+				parsedKey, err = x509.ParsePKIXPublicKey(this.publicKey.Key)
+				if err != nil {
+					return
+				}
+				var publicKey *rsa.PublicKey
+				if publicKey, ok = parsedKey.(*rsa.PublicKey); !ok {
+					return
+				}
+				saltBytes := make([]byte, 8)
+				binary.BigEndian.PutUint64(saltBytes, uint64(loginEncryptResponse.Salt))
+				digest := sha256.Sum256(append(this.verifyToken, saltBytes...))
```

Comment:
Please change the append to this:
https://github.com/LilyPad/GoLilyPad/pull/77/commits/56879ca0a3404596506fa6cba341fbc6e60065a1

Reply:
> Please change the append to this: [56879ca](https://github.com/LilyPad/GoLilyPad/commit/56879ca0a3404596506fa6cba341fbc6e60065a1)

Changed, nice catch!